src/api/ratelimits.mdx

```diff
@@ -0,0 +1,28 @@
+---
+title: Rate Limits
+sidebar_order: 5
+---
+
+Sentry rate limits every API request made to prevent abuse and resource overuse. The limit is applied to each caller per endpoint.
+
+The rate limit follows a fixed window approach. Requests are counted into time buckets, determined by the window size. If the number of requests from a caller exceeds the number of allowed requests within a window, then that request will be rejected.
```

Comment:
Should we also say that there will be a concurrent request rate limiter in the future? Might as well get people ready for it now. Also is there a killswitch for the global rate limiter? It may be good to provide instructions to self-hosted customers on how to disable it because they may not need it.

Reply:
I think it's too early to add stuff on the concurrent rate limiter. Once we have a better idea on what it will look like for users, then we can talk about adding it to the reference.

Is this the right place for self-hosted specific points? The kill switch right now is the `enforce_rate_limits` flag in `Endpoint`.

Reply:
> Is this the right place for self-hosted specific points? 

That's not a kill switch. It requires a deploy and a code change. Does sentry not have a runtime flags system in redis or flagr or something?

Reply:
We didn't want to use flagr because it would add too much latency to every request. We have to create a new system for this.

Reply:
We didn't want to use flagr because it would add too much latency to every request. We have to create a new system for this.